### PR TITLE
feat: Auto join team if already in org [CAL-2248]

### LINKED ISSUE #10401: [CAL-2248] Orgs: Invites to teams when you're already in the org should be auto-accepted
Currently invites are showing up here, even though I'm already in the org. If an admin adds you to a team, you should see it already.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2248](https://linear.app/calcom/issue/CAL-2248/orgs-invites-to-teams-when-youre-already-in-the-org-should-be-auto)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
Auto joins a team if already in the organization that the team is apart of. 

Fixes: #10401 
Fixes: CAL-2248

Loom: https://www.loom.com/share/a4993ec59f62494ab7397b05402b4135

how to test: 
invite user from organization to a team - auto join should happen. 
Invite a user not from organization - auto join should be false and the invite/onboarding flow happens.

### CODE REVIEW

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts [hunk 1]
```diff
@@ -325,3 +325,39 @@ export function getIsOrgVerified(
...
+    return {
+      autoJoined: false,
+    };
+  }
+
+  // If the user is invited to a child team, they are automatically accepted into the parent org
+  await prisma.membership.create({
+    data: {
+      userId: invitee.id,
+      teamId: team.id,
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:354]: If the user is being added to the org shouldn't the `teamId` be the team `parentId`?

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts [hunk 2]
```diff
@@ -325,3 +325,39 @@ export function getIsOrgVerified(
...
+  invitee: User;
+  role: MembershipRole;
+}) {
+  if (invitee.organizationId && invitee.organizationId !== team.parentId) {
+    return {
+      autoJoined: false,
+    };
+  }
+
+  if (!team.parentId) {
```
Comment (emrysal) [inline packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:344]: What if he joins the main organisation team?

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts [hunk 3]
```diff
@@ -69,6 +70,18 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
     } else {
       throwIfInviteIsToOrgAndUserExists(invitee, team, input.isOrg);
 
+      if (
+        await createAndAutoJoinIfInOrg({
+          invitee,
+          role: input.role,
+          team,
+        })
```
Comment (emrysal) [inline packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:78]:
```suggestion
        }).autoJoined
```

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts (hunk 1 above)
Comment (sean-brydon) [reply packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:354]: This doesnt handle to org. This is only invites to teams apart of the users org. 

Let me now if i need to comment to explain that

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts (hunk 2 above)
Comment (sean-brydon) [reply packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:344]: then parentId is null and this is skipped - we want them to have the full onboarding flow. 

if autojoined here is true we skip the invite etc

**Review by emrysal — DISMISSED** [review]
Looks good to me

**Review by emrysal — APPROVED** [review]

**Pull request merged**